SharpCompress/Writer/Zip/ZipWriter.cs

```diff
@@ -121,12 +88,13 @@ private string NormalizeFilename(string filename)
             return filename.Trim('/');
         }
 
-        private int WriteHeader(string filename, DateTime? modificationTime)
+        private int WriteHeader(string filename, DateTime? modificationTime, ZipCompressionInfo explicitCompressionInfo = null)
         {
+            var currentCompressionInfo = explicitCompressionInfo != null ? explicitCompressionInfo : this.zipCompressionInfo;

```

Comment:
I think we should handle this in only one place. How about creating a second overload of this method that takes `CompressionInfo` as a parameter:
```
private int WriteHeader(string filename, DateTime? modificationTime, CompressionInfo compressionInfo = null)
{
    var explicitCompressionInfo = compressionInfo != null ?
        new ZipCompressionInfo(compressionInfo) : this.zipCompressionInfo;
    this.WriteHeader(filename, modificationTime, explicitCompressionInfo);
}
```